paasta_tools/utils.py

```diff
@@ -113,6 +115,12 @@
 
 DEFAULT_CPU_PERIOD = 100000
 DEFAULT_CPU_BURST_ADD = 1
+LDAP_SEARCH_BASE = os.environ.get("PAASTA_SECRET_LDAP_SEARCH_BASE")
+LDAP_SEARCH_OU = os.environ.get("PAASTA_SECRET_LDAP_SEARCH_OU")
+LDAP_HOST = os.environ.get("PAASTA_SECRET_LDAP_HOST")
+LDAP_USERNAME = os.environ.get("PAASTA_SECRET_LDAP_USERNAME")
+LDAP_PASSWORD = os.environ.get("PAASTA_SECRET_LDAP_PASSWORD")
```

Comment:
I think PAASTA_SECRET_ variables are only ever set inside of containers that are run by paasta. `mark-for-deployment` doesn't run inside paasta containers (only dev boxes and Jenkins), so these won't be set.

Probably the best bet is to put this stuff in /etc/paasta, add getters to `SystemPaastaConfig`, add type definitions in `SystemPaastaConfigDict`, and then have . Though be aware that the LDAP creds will have to be world-readable for this to work.

You'd then call `load_system_paasta_config` in ldap_user_search or the calling site, or pass an existing SystemPaastaConfig object in, if available.

Reply:
I chatted with @ztm to confirm we won't be posting these new push groups as Linux groups, which prevents using `os.get_groups()`. As for having the LDAP creds be world readable, that seems to be no different than our existing setup, which we're fine with for now.

As such, I've implemented the change you proposed in the second comment above.